rai/node/rpc.cpp

```diff
@@ -1604,101 +1624,109 @@ class history_visitor : public rai::block_visitor
 			tree.put ("amount", rai::genesis_amount.convert_to<std::string> ());
 		}
 	}
-	void change_block (rai::change_block const &)
+	void change_block (rai::change_block const & block_a)
 	{
-		// Don't report change blocks
+		if (raw)
+		{
+			tree.put ("type", "change");
+			tree.put ("representative", block_a.hashables.representative.to_account ());
+		}
 	}
 	rai::rpc_handler & handler;
+	bool raw;
 	rai::transaction & transaction;
 	boost::property_tree::ptree & tree;
 	rai::block_hash const & hash;
 };
 }
 
-void rai::rpc_handler::history ()
+void rai::rpc_handler::account_history ()
 {
-	std::string hash_text (request.get<std::string> ("hash"));
 	std::string count_text (request.get<std::string> ("count"));
+	bool output_raw (*request.get_optional<bool> ("raw"));
```

Comment:
I don't think this dereference is correct.

Reply:
I'm not completely sure either. It seemed to work, but it might've just skipped an assert in release mode. I've changed it to `== true` which should definitely work, according to http://www.boost.org/doc/libs/1_56_0/libs/optional/doc/html/boost_optional/tutorial/a_note_about_optional_bool_.html